### ISSUE TITLE
No result when querying for the last record of a query

### ISSUE THREAD

**Issue by gurisko** [arangodb/arangodb#2879]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [x] 3.1
- [x] 3.2
- [ ] self-compiled devel branch

Mode:
- [ ] Cluster
- [x] Single-Server

Storage-Engine:
- [x] mmfiles
- [x] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [x] Linux 
  - [x] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

### this is an AQL-related issue:
- [ ] I'm using graph features

I'm issuing AQL via:
- [x] web interface with this browser: _Chrome_ running on this OS: _Debian 3.16.43-2 (2017-04-30) x86_64_
- [ ] arangosh
- [ ] this Driver:

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.
The AQL query in question is:

First I get the number of results using this query:
```
RETURN LENGTH (
  FOR package IN packages
    FOR depends IN (package.Depends || [])
      RETURN 1
)
```

This returns `187413`. I am going to use this knowledge to build a pagination on these data. A user requests the last record. My query:

```
FOR package IN packages
  FOR depends IN (package.Depends || [])
    LIMIT 187412, 1
    RETURN depends
```
The query results in an empty array.

The issue can be reproduced using this dataset: [Debian packages](https://github.com/arangodb/example-datasets/tree/fc4a67e399d8bc8c7e8fb224f9b6a03bac8f68af/DebianDependencyGraph)

**Comment by baslr** [comment]
The limit in your query acts on one document.

To access the last record you have to run something like this
```
FOR x IN (FOR package IN packages
  FOR depends IN (package.Depends || [])
    RETURN depends)

LIMIT 187412, 1
RETURN x
```

**Comment by jsteemann** [comment]
The result returned by AQL is still wrong. 
I have locally reproduced the problem and fixed it. Haven't found time to issue a PR for this yet.

**Comment by jsteemann** [comment]
@baslr: here's the problem demonstrated on a much simpler example: 
```
FOR doc1 IN [[1],[2],[3]] FOR doc2 IN doc1 LIMIT 0,3 RETURN doc2
[ 
  1, 
  2, 
  3 
]

FOR doc1 IN [[1],[2],[3]] FOR doc2 IN doc1 LIMIT 1,3 RETURN doc2
[ 
  2, 
  3 
]

FOR doc1 IN [[1],[2],[3]] FOR doc2 IN doc1 LIMIT 2,3 RETURN doc2
[ ]
```

**Comment by jsteemann** [comment]
Fixed in 3.2.1

**Comment by jsteemann** [comment]
Fixed in 3.1.27 as well.

**Comment by jsteemann** [comment]
3.2.1 has been released and is [available for download](https://www.arangodb.com/download-major/).
